packages/flutter/lib/src/animation/animations.dart

```diff
@@ -379,15 +379,7 @@ class ReverseAnimation extends Animation<double>
 class CurvedAnimation extends Animation<double> with AnimationWithParentMixin<double> {
   /// Creates a curved animation.
   CurvedAnimation({required this.parent, required this.curve, this.reverseCurve}) {
-    // TODO(polina-c): stop duplicating code across disposables
-    // https://github.com/flutter/flutter/issues/137435
-    if (kFlutterMemoryAllocationsEnabled) {
-      FlutterMemoryAllocations.instance.dispatchObjectCreated(
-        library: 'package:flutter/animation.dart',
-        className: '$CurvedAnimation',
-        object: this,
-      );
-    }
+    maybeDispatchObjectCreated('package:flutter/animation.dart', '$CurvedAnimation', this);
```

Comment:
Is the dart tree shaker/optimizer smart enough to remove this entire call and the `$CurvedAnimation` processing when the body of `maybeDispatchObjectCreated` is empty because kFlutterMemoryAllocationsEnabled is false? Or would we still pay a price for this line even when kFlutterMemoryAllocationsEnabled is false?

Reply:
yes, it is tree shaked when kFlutterMemoryAllocationsEnabled is false

Reply:
I just tried this program with dart2js in release mode:

```dart
void main() {
  print('MyApp');
  print('$MyApp');
}

class MyApp {}
```

And the output I got was:

```
MyApp
minified:vW
```

So I think the `'$CurvedAnimation'` interpolation may not work as we expect. But maybe it's not an issue if we only ever run this in debug mode?

Reply:
oh, interesting!
thank you for checking
we need to remove it then here and in other places